front/js/product.js

```diff
@@ -0,0 +1,51 @@
+//The function below displays the product selected by the user on the home page
+const whatProductUrl = window.location.href;
+const productUrl = new URL(whatProductUrl);
+const search_params = new URLSearchParams(productUrl.search); 
+if(search_params.has('id')) {
+  let productId = search_params.get('id');
+  fetch("http://localhost:3000/api/products")
+    .then(function(res) {
+      if (!res.ok) {
+        return res.text().then(function(text) {throw new Error(text)});
+      } else {
+        return res.json();
+      }
+    })
+    .then(function(displayIdProduct) {
+      for (let i in displayIdProduct) {
```

Comment:
for of serai plus approprier car i serai alors égale au produit courant de la boucle directement et tu pourrais te passer de faire displayIdProduct[i]